index.d.ts

```diff
@@ -45,18 +80,55 @@ export declare class Base<TOptions extends Base.Options = Base.Options> {
   ): S & {
     plugins: any[];
   } & Constructor<UnionToIntersection<ReturnTypeOf<T1> & ReturnTypeOf<T2>>>;
+
+  /**
+   * Set defaults for the constructor
+   *
+   * ```js
+   * const MyBase = Base.defaults({ version: '1.0.0', otherDefault: 'value' });
+   * const base = new MyBase({ option: 'value' }); // `version` option is not required
+   * base.options // typed as `{ version: string, otherDefault: string, option: string }`
+   * ```
+   * @remarks
+   * Ideally, we would want to make this infinitely recursive: allowing any number of
+   * .defaults({ ... }).defaults({ ... }).defaults({ ... }).defaults({ ... })...
+   * However, we don't see a clean way in today's TypeScript syntax to do so.
+   * We instead artificially limit accurate type inference to just three levels,
+   * since real users are not likely to go past that.
+   * @see https://github.com/gr2m/javascript-plugin-architecture-with-typescript-definitions/pull/57
```

Comment:
so what's interesting is that I tried adding a test with `// @ts-expect-error` for four levels of chaining, but  .... it just works? It seems to be working infinitely? 

```ts
const Test3 = Base.defaults({ version: "1.2.3", one: 1 })
  .defaults({ two: 2 })
  .defaults({ three: 3 });

const Test6 = Test3.defaults({ four: 4 })
  .defaults({ five: 5 })
  .defaults({ six: 6 })
  .plugin(withOptionsPlugin);

expectType<{
  version: string;
  one: number;
  two: number;
  three: number;
  four: number;
  five: number;
  six: number;
  // @ts-expect-error - Test3 does not have four / five / six properties
}>({ ...Test3.defaultOptions });

expectType<{
  version: string;
  one: number;
  two: number;
  three: number;
  four: number;
  five: number;
  six: number;
}>({ ...Test6.defaultOptions });
```

What is this sorcery?

Reply:
Ah I got it now, what's limited is the `.options` property on the instances. I'll add a suggestion to the tests to cover that case

Reply:
@JoshuaKGoldberg I suggest we clarify the current behavior in the README. 

TypeScript will not complain when chaining `.defaults()` calls endlessly and the static `.defaultOptions` property is set correctly. But when instantiating from a class with 4+ chained `.defaults()` calls, then only the defaults from the first 3 calls are supported. Is that correct?

Reply:
@gr2m I think that's accurate, yeah.